core/src/test/java/org/sonarsource/sonarlint/core/local/only/XodusLocalOnlyIssueStoreTests.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * SonarLint Core - Implementation
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.local.only;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.api.io.TempDir;
+import org.sonarsource.sonarlint.core.clientapi.backend.issue.IssueStatus;
+import org.sonarsource.sonarlint.core.commons.log.SonarLintLogTester;
+import org.sonarsource.sonarlint.core.tracking.LocalOnlyIssue;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.within;
+import static org.sonarsource.sonarlint.core.local.only.LocalOnlyIssueFixtures.aLocalOnlyIssueUnresolved;
+import static org.sonarsource.sonarlint.core.local.only.LocalOnlyIssueFixtures.aLocalOnlyIssueUnresolvedWithoutTextAndLineRange;
+
+class XodusLocalOnlyIssueStoreTests {
+
+  @RegisterExtension
+  SonarLintLogTester logTester = new SonarLintLogTester(true);
+
+  @TempDir
+  Path workDir;
+  @TempDir
+  Path backupDir;
+  private XodusLocalOnlyIssueStore store;
+
+  @BeforeEach
+  void setUp() throws IOException {
+    store = new XodusLocalOnlyIssueStore(backupDir, workDir);
+  }
+
+  @AfterEach
+  void tearDown() {
+    store.close();
+  }
+
+  @Test
+  void should_return_empty_when_file_path_unknown() {
+    var issues = store.load("configScopeId", "path");
+
+    assertThat(issues).isEmpty();
+  }
+
+  private void compare_issue_to_saved_one(LocalOnlyIssue savedIssue, LocalOnlyIssue localOnlyIssue, IssueStatus resolution, String comment) {
+    assertThat(savedIssue.getMessage()).isEqualTo(localOnlyIssue.getMessage());
```

Comment:
One thing I use in those kind of situations is `assertThat(savedIssue).usingRecursiveComparison().isEqualTo(localOnlyIssue)`. Even if there is no equals, it will compare field by field

Reply:
Thanks, I didn't know this! I added `ignoringFieldsOfTypes(LocalOnlyIssueResolution.class)` because it is added when storing the issue